resource/systemd/unit/unit.go

```diff
@@ -0,0 +1,543 @@
+			return &resource.Status{
+				Status: statusMsg,
+				Output: []string{statusMsg},
+			}, nil
+		}
+		if state.Equal(systemd.UFSStatic) {
+			changes, err := dbusConn.MaskUnitFiles([]string{unitName}, false, true)
+			if err != nil {
+				status := &resource.Status{
+					WarningLevel: resource.StatusFatal,
+					Status:       err.Error(),
+					Output:       []string{err.Error()},
+				}
+				return status, err
+			}
+			statusMsg := fmt.Sprintf("unit %q is maksed", unitName)
+			status := &resource.Status{
+				Status: statusMsg,
+				Output: []string{
+					statusMsg,
+					fmt.Sprintf("symlink for %q was %q. link between %q and %q", unitName, changes[0].Type, changes[0].Filename, changes[0].Destination),
+				},
+			}
+			return status, err
+		}
+
+		_, err := dbusConn.DisableUnitFiles([]string{unitName}, state.IsRuntimeState())
+		if err != nil {
+			status := &resource.Status{
+				WarningLevel: resource.StatusFatal,
+				Status:       err.Error(),
+				Output:       []string{err.Error()},
+			}
+			return status, err
+		}
+		statusMsg := fmt.Sprintf("unit %q has been disabled", unitName)
+		return &resource.Status{
+			Status: statusMsg,
+			Output: []string{
+				statusMsg,
+			},
+		}, nil
+	/*///////////////////////////////////
+	*Linking
+	 */ ///////////////////////////////////
+	case systemd.UFSLinked:
+		fallthrough
+	case systemd.UFSLinkedRuntime:
+		// must change from linked to linked-runtime or vice versa
+		if state.IsLinked() {
+			_, err := dbusConn.DisableUnitFiles([]string{unitName}, state.IsRuntimeState())
+			if err != nil {
+				status := &resource.Status{
+					WarningLevel: resource.StatusFatal,
+					Status:       err.Error(),
+					Output:       []string{err.Error()},
+				}
+				return status, err
+			}
+
+			changes, e := dbusConn.LinkUnitFiles([]string{t.Name}, t.UnitFileState.IsRuntimeState(), true)
+			if e != nil {
+				status := &resource.Status{
+					WarningLevel: resource.StatusFatal,
+					Status:       e.Error(),
+				}
+				return status, e
+			}
+			statusMsg := fmt.Sprintf("unit %q has been linked", unitName)
+			diffMsg := fmt.Sprintf("unit %q does not exist", unitName)
+			diff := resource.TextDiff{Default: diffMsg, Values: [2]string{diffMsg, fmt.Sprintf("unit %q is linked", unitName)}}
+			return &resource.Status{
+				Status:       statusMsg,
+				WarningLevel: resource.StatusNoChange,
+				WillChange:   false,
+				Differences:  map[string]resource.Diff{unitName: diff},
+				Output: []string{
+					statusMsg,
+					fmt.Sprintf("file %q has been linkded to %q", changes[0].Filename, changes[0].Destination),
+				},
+			}, nil
+		}
+
+	}
+
+	statusMsg := fmt.Sprintf("unit %q is in unknown state, %s", unitName, state)
+	return &resource.Status{
+		Status:       statusMsg,
+		WarningLevel: resource.StatusFatal,
+		Output:       []string{statusMsg},
+	}, fmt.Errorf(statusMsg)
+}
+
+func (t *Unit) StartOrStop(dbusConn *dbus.Conn) (*resource.Status, error) {
```

Comment:
It's less boilerplate code to have them be the same function. Starting Or Stopping is just a one line difference.